src/main/webapp/app/course/manage/course-storage.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Course } from 'app/entities/course.model';
+import { SubjectObservablePair } from 'app/utils/rxjs.utils';
+
+@Injectable({ providedIn: 'root' })
+export class CourseStorageService {
+    private storedCourses: Course[] = [];
+
+    private readonly courseUpdates: Map<number, SubjectObservablePair<Course>> = new Map();
+
+    setCourses(courses: Course[]) {
+        this.storedCourses = courses;
+    }
+
+    getCourse(courseId: number) {
+        return this.storedCourses.find((course) => course.id === courseId);
+    }
+
+    updateCourse(course: Course) {
```

Comment:
Why are `updateCourse` and `notifyCourseUpdatesSubscribers` two different methods? Shouldn't every call to update als notify the subscribers?

Reply:
I am not entirely the idea here was. I consolidated all methods about getting and setting the course from the `course-mangement.service.ts` and `course-score-calculation.service.ts` into the `course-storage.service.ts` which makes it obvious that this is a little weird. I guess this way you could just update the course in one component and then other components would not be updated until they themselves call `getCourse()` again? Not sure if this is necessary though. I did a quick refactoring to only use `updateCourse()` and always notify subscribers in that method and it seems to not have broken anything.